src/main/scala/com/exasol/cloudetl/kinesis/KinesisShardDataImporter.scala

```diff
@@ -66,14 +70,19 @@ object KinesisShardDataImporter {
     getShardIteratorRequest
   }
 
+  @SuppressWarnings(Array("org.wartremover.warts.OptionPartial"))
   private[kinesis] def getRecords(
     amazonKinesis: AmazonKinesis,
-    shardIteratorRequest: GetShardIteratorRequest
+    shardIteratorRequest: GetShardIteratorRequest,
+    limit: Option[Int]
   ): List[Record] = {
     val shardIteratorResult = amazonKinesis.getShardIterator(shardIteratorRequest)
     val shardIterator = shardIteratorResult.getShardIterator
     val getRecordsRequest = new GetRecordsRequest
     getRecordsRequest.setShardIterator(shardIterator)
+    if (limit.isDefined) {
+      getRecordsRequest.setLimit(limit.get)
+    }
```

Comment:
Let's use the `Option.fold` and remove the suppression.

```scala
limitOpt.fold({}) { limitValue =>
getRecordsRequest.setLimit(limitValue)
}
```
The function inside `fold` is None case, so empty unit. The second function is for Some case, use the value.

It is long running discussion which is better, `getOrElse`, `isDefined/get`, pattern-matching, `fold`. Using fold is more preferred.

You can read more:
- https://www.originate.com/idiomatic-scala-your-options-do-not-match/
- https://kwangyulseo.com/2014/05/21/scala-option-fold-vs-option-mapgetorelse/
- https://old.reddit.com/r/scala/comments/449td4/fold_or_pattern_matching_with_an_option/

Reply:
done